src/qibolab/_core/instruments/qm/program/instructions.py

```diff
@@ -52,7 +52,10 @@ def _play_single_waveform(
     if acquisition is not None:
         acquisition.measure(op)
     else:
-        qua.play(op, element, duration=parameters.duration)
+        if parameters.duration is not None:
+            qua.play(parameters.interpolated_op, element, duration=parameters.duration)
```

Comment:
Ok, I guess you're not entering here if the sweeper is a non-interpolated duration sweeper (since you should be using `_play_multiple_waveforms`).

Still, would you leave a comment to clarify that's the case? Or an assertion/exception with a message, if you prefer. Just to declare that you should not end up here with the other sweeper (such that, in case it happens, I may also remember it :P)

Reply:
Added some assertions in 6cefd7e4dbbfabb27f03a2322b11f4ff9cf5e36e.